com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -189,6 +188,34 @@ public UniversalRenderer(UniversalRendererData data) : base(data)
             m_FinalDepthCopyPass = new CopyDepthPass(RenderPassEvent.AfterRendering + 9, m_CopyDepthMaterial);
 #endif
 
+            if(DebugHandler != null)
+            {
+                // Hook in the debug-render where appropriate...
+                m_RenderOpaqueForwardPass.DebugHandler = DebugHandler;
+                m_FinalBlitPass.DebugHandler = DebugHandler;
+
+                if(m_RenderOpaqueForwardOnlyPass != null)
+                {
+                    m_RenderOpaqueForwardOnlyPass.DebugHandler = DebugHandler;
+                }
+                if(m_RenderTransparentForwardPass != null)
+                {
+                    m_RenderTransparentForwardPass.DebugHandler = DebugHandler;
+                }
+                if(m_DrawSkyboxPass != null)
+                {
+                    m_DrawSkyboxPass.DebugHandler = DebugHandler;
+                }
+                if(m_PostProcessPasses.postProcessPass != null)
+                {
+                    m_PostProcessPasses.postProcessPass.DebugHandler = DebugHandler;
+                }
+                if(m_PostProcessPasses.finalPostProcessPass != null)
+                {
+                    m_PostProcessPasses.finalPostProcessPass.DebugHandler = DebugHandler;
+                }
+            }
```

Comment:
DebugHandler is exposed throught the renderer, maybe we should access via renderer instead of setting access per pass.

Reply:
Yes, should be possible. If you agree, this could be for the follow-up PR.

Reply:
https://jira.unity3d.com/browse/XPIPELINE-157